File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None: return None

        stk = []
        while head != None:
            stk.append(head)
            head = head.next
        
        new_head = stk.pop()
        cur = new_head
        while len(stk) > 0:
            cur.next = stk.pop()
            cur = cur.next
        cur.next = None
        
        return new_head

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy_head = new ListNode();
        ListNode start = dummy_head;
        while(list1 != null && list2 != null) {
            if(list1.val <= list2.val) {
                dummy_head.next = list1;
                list1 = list1.next;
            }
            else {
                dummy_head.next = list2;
                list2 = list2.next;
            }
            dummy_head = dummy_head.next;
        }
        if(list1 != null) {
            dummy_head.next = list1;
        }
        else if(list2 != null) {
            dummy_head.next = list2;
        }

        return start.next;
    }
}